deepchecks/nlp/input_validations.py

```diff
@@ -231,3 +235,197 @@ def compare_dataframes(
         difference = None
 
     return DataframesComparison(common, difference)
+
+
+def _validate_text_classification(
```

Comment:
I would imagine that the difference between these and tabular _validate_classification would be that tabular has also binary. Is there a way to unify this multiclass logic with the tabular multiclass logic?

Reply:
> Is there a way to unify this multiclass logic with the tabular multiclass logic?

and Yes and No/

you know, IMO, it is one of those cases when code indeed has some similarities but differ in details and it is  not obvious how to separate those similarities into generic enough function, and if you will try you will end up with a code that is less readable and is harder to understand

for example, I do not find error messages in the `_validate_text_classification` to be suited for the tabular case, so should a unified function accept error messages templates?/

IMO, not worth overcomplicating very simple code just to reduce a few lines